src/Categories/Multi/Category/Cartesian.agda

```diff
@@ -0,0 +1,115 @@
+{-# OPTIONS --without-K --safe #-}
+
+-- Idea:
+-- Whereas ordinary multicategories are based around the notion of substitution
+-- from noncommutative linear (Lambek) logic, Cartesian multicategories are
+-- based around the notion of substitution from intuitionistic logic, which is
+-- a lot simpler. A representable Cartesian multicategory is exactly a
+-- Cartesian category.
+--
+-- See https://ncatlab.org/nlab/show/cartesian+multicategory, particularly the
+-- section “Alternative presentation”
+-- (https://ncatlab.org/nlab/show/cartesian+multicategory#alternative_presentation).
+
+module Categories.Multi.Category.Cartesian where
+
+open import Data.List hiding ([_])
+open import Data.List.Membership.Propositional
+open import Data.Wrap
+open import Level
+open import Relation.Binary using (Rel; Setoid; IsEquivalence)
+import Relation.Binary.Reasoning.Setoid as SetoidR
+
+infix 4 [_]_≈ᵉ_
+
+Env : ∀ {o ℓ} {X : Set o} (V : X → Set ℓ) (Δ : List X) → Set (o ⊔ ℓ)
+Env V Δ = ∀ {y} → y ∈ Δ → V y
+
+[_]_≈ᵉ_ : ∀ {o ℓ e} {X : Set o} {V : X → Set ℓ} {Δ} →
+  (∀ {x} → Rel (V x) e) → Rel (Env V Δ) (o ⊔ e)
+[_]_≈ᵉ_ = Wrap λ _≈_ ρ σ → ∀ {y} (i : y ∈ _) → ρ i ≈ σ i
+
+record CartesianMultiCategory (o ℓ e : Level) : Set (suc (o ⊔ ℓ ⊔ e)) where
+  infix 4 _≈_ _≈ˢ_ _⇒_ _⇒ˢ_
```

Comment:
For example, does the superscript 's' *mean* 'simultaneous'? That would be good to write down.

Reply:
Yes, though maybe there could be a better notation (`_⇒*_` might work, with the `*` being a mnemonic for lists and multiplicity, though maybe also clashing with reflexive-transitive closure). I'll note it down, in any case.

Reply:
I agree that `*` is too overloaded and might lead to unnecessary confusion. Perhaps `∀` in the symbol? It's almost too bad you've already used `⇒Π`.